pilot/pkg/model/service.go

```diff
@@ -234,6 +234,39 @@ func (instance *ServiceInstance) DeepCopy() *ServiceInstance {
 	}
 }
 
+// a custom comparison of foreign service instances based on the fields that we need
+// i.e. excluding the ports. Returns true if equal, false otherwise.
+func CompareForeignServiceInstances(first, second *ServiceInstance) bool {
```

Comment:
nit: Change CompareForeignServices -> ForeignSeviceInstancesEqual to be more clear?